### DISCUSSION TITLE
Unhandled runtime error when clicking on new cat

### DISCUSSION THREAD

**Discussion by JanaS87** [neuefische/questions#681] [Web] [answered]
When I create a new cat, everything works fine. I switch back to the overview (with the back link) and the cat will be displayed in the cat list.
When I click on that cat I get this runtime error:
![Bildschirmfoto 2023-08-18 um 15 47 11](https://github.com/neuefische/questions/assets/136725332/ff471664-95dc-4260-b67f-c90e39e406d7)

My PR: https://github.com/JanaS87/capstone-project/pull/8
(add new cat -> fill the form -> save -> back -> click on new cat)

How can I solve this problem? I don't know where I have to look at.

Maybe someone here knows what to do.

Thank you in advance!

**Comment by Kheirah** [comment, marked as answer]
The current `catdata.js` does not have an array for the properties `goodAcceptance` and `badAcceptance`.
```js
export const cats = [
  {
    id: "1",
    name: "Luna",
    age: 5,
    allergies: "",
    diseases: "",
    intolerances: "",
    goodAcceptance: "Wild Freedom Instinctive", //string; not an array
    badAcceptance: "Sheba", //string; not an array
  },
...
```
However, when you create a new cat, both properties for good and bad cat food **_are arrays:_**
```js
function handleSubmit(event) {
    event.preventDefault();

    // generate new ID
    const newId = uid();

    // create new cat object at the end of the array
    const newCat = {
      ...cat,
      id: newId,
      goodAcceptance: addedGoodFood, //this is an array of objects with keys { value, label }
      badAcceptance: addedBadFood, //this is an array of objects with keys { value, label }
    };
...
```
In your `pages/cats/[id].js` you are accessing both properties as if they weren't arrays (and that's true for your `catdata.js` but not true for any newly added cat):
```js
<h4>Good Acceptance: </h4>
<p>[{cat.goodAcceptance}]</p> //when it's an array, you'd have to map over it
<h4>Bad Acceptance: </h4>
<p>[{cat.badAcceptance}]</p> //the same is true here
...
```

**Reply by JanaS87** [reply]
Thank you, that helped me a lot!
